docs/adr/014-rollover-2.0.md

```diff
@@ -0,0 +1,91 @@
+   - Every day at 1200 UTC.
+   - Every day at 1600 UTC.
+   - Every day at 2000 UTC.
+   - Every day at 0000 UTC.
+
+Running rollover at exactly these points makes implementations easier because they can assume that well-behaved peer will execute the protocol at these points in time.
+
+It makes UI design easier because we can clearly state, when the next funding event happens, and it does not depend on the maker that we are connected to.
+
+Another benefit of fixed funding periods is that our oracle actor only needs to fetch attestations for specific points in time and not every hour.
+
+### 2. Each party should start the protocol with an expectation of the funding rate
+
+In the future, the funding rate will be determined by the market.
+As such, no party should be specifying the funding rate as part of the protocol.
+
+### 3. Maker streams current funding rate to taker
+
+To simulate a market whilst we don't yet have one, introduce a stream of funding rates from the maker to the taker.
+This stream can be used to display the current funding rate in the UI.
+Once we have a market, this component will be removed.
+
+### 4. Enforce agreement of the funding rate through the blockchain protocol
+
+With both parties entering the next funding period with a specific funding rate, the protocol will simply fail due to signature verification errors if the funding rates don't match.
+This should be sufficient for an MVP of this new protocol.
+More elaborate detection of what the cause of the failure was can easily be added on top.
+
+### 5. Remove approval (accept / decline) from the protocol
+   
+With (1) and (2) in mind, there is really no point in an external approval step for rollover.
+As long as the CFD's is not committed on-chain, rollover will occur.
+To opt out of rollover, either party can commit the CFD on-chain which will result in a payout at the end of the funding period.
+
+### 6. Connection listener sends initial trigger message
+
+To make the system more predictable, we should have an dedicated trigger for the protocol.
+One way of ensuring this is deterministic, is to check whether a peer is the dialer or the listener in a connection.
+
+In our current setup, the maker is the listener.
+If the have the maker send out the initial trigger message, we have _some_ control over how many rollovers we want to do in parallel.
+If the taker would send the intial trigger message and we don't build in some kind of randomness in when to send it, rollover might end up being a DoS attack against the maker.
+
+## Considerations
+
+1. The new protocol requires being online at a very specific point in time.
+   
+   This requirement is already implied by the fact that we have an off-chain channel construct and we need to be constantly online to monitor for the broadcast of old commit transactions.
+
+## Open questions
+
+1. Should we avoid the in-efficiency in doing a rollover when both parties don't align on the funding rate?
```

Comment:
Just let rollover fail at this stage, if that's what this question is getting at.

Reply:
That would have been my initial idea as well but @bonomat rightfully pointed out that it creates an in-efficiency where we do unnecessary computations that end up not working.

I am torn here. On the one hand, we should obviously be as efficient as possible. However, there are really only two scenarios in which this could happen (I think):

- There is a bug in our software where both parties compute the wrong funding rate (assuming it somehow consensus / market-driven)
- We have a rouge peer that wants to DoS us by making us do rollovers that purposely fail

An initial check could circumvent the first problem but not the second because a rouge peer can just send on value in the initial check and then send transactions that are computed using a different funding rate.

Considering all this, I think the value of a check is pretty small. In normal operation, all peers should always compute the same funding rate and in the case of malicious peers, we would need to deal with that differently anyway (like ban / disconnect from them if they show funny behaviour).